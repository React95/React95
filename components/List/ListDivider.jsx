import styled from 'styled-components';

const Divider = styled.li`
  background-color: #a2a5a8;
  height: 1px;
  border-top: 1px solid #868a8e;
  border-bottom: 1px solid white;

  width: 98%;
  margin-left: 2px;
`;

export default Divider;
