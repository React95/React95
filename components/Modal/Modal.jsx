import React, { useState } from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import Draggable from 'react-draggable';
import Btn from '../shared-style/Btn';
import Button from '../Button';
import Icon from '../Icon';

const ModalWrapper = styled.div`
  display: flex;
  flex-direction: column;

  position: fixed;

  padding: 2px 2px 8px;
  ${({ width, height }) => `width: ${width}px; height: ${height}px;`}

  top: 50px;

  background-color: #c3c7cb;

  box-shadow: inset 1px 1px 0px 1px #ffffff, inset 0 0 0 1px #868a8e,
    1px 1px 0 1px #000;
`;

const TitleBar = styled.div`
  height: 18px;
  margin-bottom: 2px;

  background-color: #00007f;
  color: white;
  padding: 2px 2px 0;

  display: flex;
`;

const Title = styled.div`
  flex-grow: 1;
`;

const OptionsBox = styled.ul`
  list-style: none;
  padding: 0;
  margin: 0;

  display: flex;
`;

const Option = styled(Btn)`
  margin-right: 2px;
  padding: 0;

  width: 17px;
  height: 14px;
  min-width: 0;

  font-size: 10px;

  &:last-child {
    margin-right: 0;
  }

  &:active {
    padding: 1px 0 0 1px;

    outline: none;
  }

  &:focus {
    box-shadow: inset 1px 1px 0px 1px #ffffff, inset -1px -1px 0px 1px #868a8e;
  }
`;

const Content = styled.div`
  flex-grow: 1;
  display: flex;
  flex-direction: column;

  padding: 6px;
`;

const ButtonWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: ${props => props.buttonsAlignment};

  padding: 0 6px 6px 6px;

  & ${Btn} {
    margin-right: 6px;
    min-width: 70px;

    &:last-child {
      margin-right: 0;
    }
  }
`;

const MenuWrapper = styled.ul`
  display: flex;
  flex-direction: row;

  list-style: none;
  margin: 0;
  padding-left: 0;
  padding-bottom: 3px;

  border-bottom-style: solid;
  border-bottom-width: 1px;
  border-bottom-color: #848284;

  box-shadow: 0 1px 0 0 #e6e6e6;
`;

const MenuItem = styled.li`
  position: relative;
  padding-left: 6px;
  padding-right: 6px;

  user-select: none;

  ul {
    position: absolute;
    left: 0;
    color: #000;
  }

  ${({ active }) => active
    && `
      background-color: #00007f;
      color: #FFF;
    `};
`;

const Modal = ({
  closeModal,
  title,
  children,
  buttons,
  icon,
  menu,
  top,
  left,
  buttonsAlignment,
  defaultPosition,
  width,
  height,
}) => {
  const [menuOpened, setMenuOpened] = useState('');

  const iconStyle = {
    width: 15,
    height: 13,
    style: {
      marginRight: '4px',
    },
  };

  const position = {
    top,
    left,
  };

  return (
    <React.Fragment>
      <Draggable handle=".draggable" defaultPosition={defaultPosition}>
        <ModalWrapper style={position} width={width} height={height}>
          <TitleBar className="draggable">
            {icon && <Icon name={icon} {...iconStyle} />}
            <Title>{title}</Title>
            <OptionsBox>
              <Option>?</Option>
              <Option onClick={closeModal}>x</Option>
            </OptionsBox>
          </TitleBar>

          {menu.length > 0 && (
            <MenuWrapper>
              {menu.map(({ name, list }) => {
                const active = menuOpened === name;
                return (
                  <MenuItem
                    key={name}
                    onMouseDown={() => setMenuOpened(name)}
                    active={active}
                  >
                    {name}
                    {active && list}
                  </MenuItem>
                );
              })}
            </MenuWrapper>
          )}

          <Content onClick={() => setMenuOpened('')}>{children}</Content>
          {buttons.length > 0 && (
            <ButtonWrapper buttonsAlignment={buttonsAlignment}>
              {buttons.map((button, index) => (
                <Button
                  key={index}
                  onClick={button.onClick}
                  value={button.value}
                />
              ))}
            </ButtonWrapper>
          )}
        </ModalWrapper>
      </Draggable>
    </React.Fragment>
  );
};

Modal.displayName = 'Modal';

Modal.propTypes = {
  icon: PropTypes.string,
  closeModal: PropTypes.func,
  title: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
  buttons: PropTypes.arrayOf(
    PropTypes.shape({
      value: PropTypes.string.isRequired,
      onClick: PropTypes.func,
    }),
  ),
  menu: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      list: PropTypes.oneOfType([
        PropTypes.arrayOf(PropTypes.node),
        PropTypes.node,
      ]).isRequired,
    }),
  ),
  buttonsAlignment: PropTypes.oneOf(['center', 'flex-start', 'flex-end']),
  defaultPosition: PropTypes.shape({
    x: PropTypes.number.isRequired,
    y: PropTypes.number.isRequired,
  }),
  width: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  height: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
};

Modal.defaultProps = {
  icon: '',
  title: 'Modal',
  buttonsAlignment: 'flex-end',
  chidren: null,
  defaultPosition: { x: 0, y: 0 },
  buttons: [],
  menu: [],
  width: 300,
  height: 200,
  closeModal: () => {},
};

export default Modal;
