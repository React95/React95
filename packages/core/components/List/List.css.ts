import { style, globalStyle } from '@vanilla-extract/css';
import { calc } from '@vanilla-extract/css-utils';
import { contract } from '../themes/contract.css';

import rightcaret from '../shared/imgs/rightcaret.svg';

export const listItem = style({
  position: 'relative',
  display: 'flex',
  alignItems: 'center',
  userSelect: 'none',
  margin: contract.space[0],
  paddingBlock: contract.space[2],
  paddingInline: contract.space[6],
  color: contract.colors.materialText,
  selectors: {
    '&:hover': {
      backgroundColor: contract.colors.headerBackground,
      color: contract.colors.materialTextInvert,
    },
    '&:not(:has(svg))': {
      paddingInlineStart: '26px',
    },
    '&:has(ul):after': {
      position: 'absolute',
      width: contract.space[5],
      height: contract.space[8],
      right: contract.space[8],
      content: "''",
      backgroundColor: contract.colors.materialText,
      maskImage: `url('${rightcaret}')`,
      maskPosition: 'center center',
      maskSize: `${contract.space[5]} ${contract.space[8]}`,
      maskRepeat: 'no-repeat',
    },
    '&:has(ul):hover:after': {
      backgroundColor: contract.colors.materialTextInvert,
    },
  },
});

globalStyle(`${listItem} svg`, {
  marginRight: contract.space[10],
  minWidth: contract.space[16],
});

globalStyle(`${listItem} ul`, {
  display: 'none',
  position: 'absolute',
  top: calc.negate(contract.space[2]),
  left: '97%',
  color: contract.colors.materialText,
  zIndex: contract.zIndices.taskbar,
});

globalStyle(`${listItem}:has(ul):hover > ul`, {
  display: 'block',
});

export const divider = style({
  height: contract.space[1],
  borderTopStyle: 'solid',
  borderTopWidth: contract.space[1],
  borderTopColor: contract.colors.borderDark,
  borderBottomWidth: contract.space[1],
  borderBottomStyle: 'solid',
  borderBottomColor: contract.colors.borderLightest,
  width: '98%',
  marginLeft: contract.space[2],
});

export const list = style({
  backgroundColor: contract.colors.material,
  padding: contract.space[2],
  border: 'none',
  margin: 0,
  listStyle: 'none',
  boxShadow: contract.shadows.out,
});
