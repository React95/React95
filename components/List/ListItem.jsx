import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Icon from '../Icon';
import List from './List';

import rightcaret from '../../assets/pattern/rightcaret.svg';

const Item = styled.li`
  position: relative;

  display: flex;
  align-items: center;
  margin: 0;
  padding: 2px 6px;

  i {
    margin-right: 10px;
  }

  &:hover {
    background-color: #00007f;
    color: white;
  }

  ul {
    display: none;
    position: absolute;
    top: -2px;
    left: 97%;
    color: #000;

    i {
      width: 18px;
      height: 20px;
      margin-right: 6px;
    }
  }

  svg {
    fill: white;
  }

  ${({ icon }) => !icon && 'padding-left: 26px;'};

  ${({ hasList }) => hasList
    && `
    &:after {
      position: absolute;
      content: '';
      background-color: #000;
      mask-image: url('${rightcaret}');
      mask-position: center center;
      mask-size: 5px 8px;
      mask-repeat: no-repeat;
      width: 5px;
      height: 8px;
      right: 8px;
    }

    &:hover {
      &:after {
        background-color: #FFF;
      }

      ul {
        display: block;
      }
    }
    `};
`;

const ListItem = ({
  icon, children, onClick, rest,
}) => (
  <Item
    {...rest}
    onClick={onClick}
    icon={icon}
    hasList={React.Children.map(
      children,
      children => children.type === List,
    ).some(child => child)}
  >
    {icon && <Icon name={icon} />}
    {children}
  </Item>
);
export default ListItem;
