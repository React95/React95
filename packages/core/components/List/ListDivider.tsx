import React, { forwardRef } from 'react';

import { Frame, FrameProps } from '../Frame/Frame';
import { divider } from './List.css';
import cn from 'classnames';

export type DividerProps = Omit<FrameProps<'li'>, 'as'>;

export const Divider = forwardRef<HTMLLIElement, DividerProps>((rest, ref) => (
  <Frame {...rest} ref={ref} className={cn(divider, rest.className)} as="li" />
));

Divider.displayName = 'List.Divider';
