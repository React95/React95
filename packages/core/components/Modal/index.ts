import { Modal } from './Modal';
import ModalContext from './ModalContext';
import ModalProvider from './ModalProvider';

export { Modal, ModalContext, ModalProvider };
