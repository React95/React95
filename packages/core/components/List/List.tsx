import React, { forwardRef } from 'react';
import cn from 'classnames';

import { ListItem } from './ListItem';
import { Divider } from './ListDivider';
import { Frame, FrameProps } from '../Frame/Frame';
import { list } from './List.css';

export type ListProps = Omit<FrameProps<'ul'>, 'as'>;

const ListRenderer = forwardRef<HTMLUListElement, ListProps>((rest, ref) => (
  <Frame {...rest} ref={ref} className={cn(list, rest.className)} as="ul" />
));

export type IListProps = typeof ListRenderer & {
  Item: typeof ListItem;
  Divider: typeof Divider;
};

export const List: IListProps = Object.assign(ListRenderer, {
  Item: ListItem,
  Divider: Divider,
});
