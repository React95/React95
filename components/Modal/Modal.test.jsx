import React from 'react';
import Modal from './Modal';
import List from '../List';
import { render, fireEvent } from 'react-testing-library';

describe('<Modal />', () => {
  describe('Snapshots', () => {
    it('should match snapshot', () => {
      const { container } = render(
        <Modal
          icon="bat"
          title="file.bat"
          buttons={[{ value: 'Ok' }, { value: 'Cancel' }]}
          menu={[
            {
              name: 'File',
              list: (
                <List>
                  <List.Item>Exit</List.Item>
                </List>
              ),
            },
            {
              name: 'Edit',
              list: (
                <List>
                  <List.Item>Copy</List.Item>
                </List>
              ),
            },
          ]}
        >
          Hello
        </Modal>
      );
      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with buttonsAligment prop equals center', () => {
      const { container } = render(
        <Modal
          icon="bat"
          title="file.bat"
          buttons={[{ value: 'Ok' }, { value: 'Cancel' }]}
          buttonsAlignment="center"
        >
          Hello
        </Modal>
      );
      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with buttonsAligment prop equals flex-start', () => {
      const { container } = render(
        <Modal
          icon="bat"
          title="file.bat"
          buttons={[{ value: 'Ok' }, { value: 'Cancel' }]}
          buttonsAlignment="flex-start"
        >
          Hello
        </Modal>
      );
      expect(container).toMatchSnapshot();
    });
  });

  describe('closeModal prop', () => {
    it('should call closeModal when Modal close button is clicked', async () => {
      const closeModalMock = jest.fn();
      const { getByText } = render(
        <Modal icon="bat" title="file.bat" closeModal={closeModalMock}>
          Hello
        </Modal>
      );

      fireEvent.click(getByText('x'));

      expect(closeModalMock).toHaveBeenCalled();
    });
  });

  describe('Modal action buttons', () => {
    it('should display button text correctly', () => {
      const buttonText = 'button text';
      const { container } = render(
        <Modal icon="bat" title="file.bat" buttons={[{ value: buttonText }]}>
          Hello
        </Modal>
      );

      expect(
        container.querySelector(`button[value='${buttonText}']`).textContent
      ).toBe(buttonText);
    });

    it('should display more than one button', () => {
      const { getAllByText } = render(
        <Modal
          icon="bat"
          title="file.bat"
          buttons={[
            { value: 'button' },
            { value: 'button' },
            { value: 'button' },
          ]}
        >
          Hello
        </Modal>
      );

      expect(getAllByText('button').length).toBe(3);
    });

    it('should call onClick function when Modal action button is clicked', () => {
      const onClickMock = jest.fn();
      const { getByText } = render(
        <Modal buttons={[{ value: 'Ok', onClick: onClickMock }]}>Hello</Modal>
      );

      fireEvent.click(getByText('Ok'));

      expect(onClickMock).toHaveBeenCalled();
    });
  });

  describe('Menus', () => {
    it('should display menu name correctly', () => {
      const { container } = render(
        <Modal
          menu={[
            {
              name: 'Menu Text',
              list: (
                <List>
                  <List.Item>Exit</List.Item>
                </List>
              ),
            },
          ]}
        >
          Hello
        </Modal>
      );

      expect(container.querySelector('li').textContent).toBe('Menu Text');
    });

    it('should display menu list correctly', () => {
      const { container, getByText } = render(
        <Modal
          menu={[
            {
              name: 'Edit',
              list: (
                <List>
                  <List.Item>Exit</List.Item>
                </List>
              ),
            },
          ]}
        >
          Hello
        </Modal>
      );

      fireEvent.mouseDown(getByText('Edit'));

      expect(getByText('Edit').querySelector('li').textContent).toBe('Exit');
    });
  });
});
