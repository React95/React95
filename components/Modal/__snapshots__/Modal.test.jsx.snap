// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> Snapshots should match snapshot 1`] = `
.c11 {
  background-color: #c3c7cb;
  padding: 7px 20px 5px;
  border: none;
  font-size: 12px;
  box-shadow: inset 1px 1px 0px 1px #ffffff,inset 0 0 0 1px #868a8e,1px 1px 0 0px #000;
}

.c11:disabled {
  color: #868a8e;
}

.c11:focus {
  outline: none;
  box-shadow: inset 1px 1px 0px 1px #ffffff,inset -0.5px -0.5px 0px 1px #868a8e,1px 1px 0 1px #000;
  outline: 1px dotted #000;
  outline-offset: -5px;
}

.c11:active {
  padding: 8px 20px 5px;
  outline: 1px dotted #000;
  outline-offset: -5px;
  box-shadow: inset 0 0 0 1px #868a8e,0 0 0 1px #000;
}

.c2 {
  display: block;
  width: 15px;
  height: 13px;
  background-image: url('test-file-stub');
  background-repeat: no-repeat;
  background-position: center;
  background-size: contain;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: fixed;
  padding: 2px 2px 8px;
  width: 300px;
  height: 200px;
  top: 50px;
  background-color: #c3c7cb;
  box-shadow: inset 1px 1px 0px 1px #ffffff,inset 0 0 0 1px #868a8e,1px 1px 0 1px #000;
}

.c1 {
  height: 18px;
  margin-bottom: 2px;
  background-color: #00007f;
  color: white;
  padding: 2px 2px 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c4 {
  list-style: none;
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c6 {
  background-color: #c3c7cb;
  padding: 7px 20px 5px;
  border: none;
  font-size: 12px;
  box-shadow: inset 1px 1px 0px 1px #ffffff,inset 0 0 0 1px #868a8e,1px 1px 0 0px #000;
  margin-right: 2px;
  padding: 0;
  width: 17px;
  height: 14px;
  min-width: 0;
  font-size: 10px;
}

.c6:disabled {
  color: #868a8e;
}

.c6:focus {
  outline: none;
  box-shadow: inset 1px 1px 0px 1px #ffffff,inset -0.5px -0.5px 0px 1px #868a8e,1px 1px 0 1px #000;
  outline: 1px dotted #000;
  outline-offset: -5px;
}

.c6:active {
  padding: 8px 20px 5px;
  outline: 1px dotted #000;
  outline-offset: -5px;
  box-shadow: inset 0 0 0 1px #868a8e,0 0 0 1px #000;
}

.c6:last-child {
  margin-right: 0;
}

.c6:active {
  padding: 1px 0 0 1px;
  outline: none;
}

.c6:focus {
  box-shadow: inset 1px 1px 0px 1px #ffffff,inset -1px -1px 0px 1px #868a8e;
}

.c9 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 6px;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  padding: 0 6px 6px 6px;
}

.c10 .c5 {
  margin-right: 6px;
  min-width: 70px;
}

.c10 .c5:last-child {
  margin-right: 0;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  list-style: none;
  margin: 0;
  padding-left: 0;
  padding-bottom: 3px;
  border-bottom-style: solid;
  border-bottom-width: 1px;
  border-bottom-color: #848284;
  box-shadow: 0 1px 0 0 #e6e6e6;
}

.c8 {
  position: relative;
  padding-left: 6px;
  padding-right: 6px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c8 ul {
  position: absolute;
  left: 0;
  color: #000;
}

<div>
  <div
    class="react-draggable c0"
    height="200"
    style="transform: translate(0px,0px);"
    width="300"
  >
    <div
      class="draggable c1"
    >
      <i
        class="c2"
        height="13"
        name="bat"
        style="margin-right: 4px;"
        width="15"
      />
      <div
        class="c3"
      >
        file.bat
      </div>
      <ul
        class="c4"
      >
        <button
          class="c5 c6"
        >
          ?
        </button>
        <button
          class="c5 c6"
        >
          x
        </button>
      </ul>
    </div>
    <ul
      class="c7"
    >
      <li
        class="c8"
      >
        File
      </li>
      <li
        class="c8"
      >
        Edit
      </li>
    </ul>
    <div
      class="c9"
    >
      Hello
    </div>
    <div
      class="c10"
    >
      <button
        class="c5 c11"
        value="Ok"
      >
        Ok
      </button>
      <button
        class="c5 c11"
        value="Cancel"
      >
        Cancel
      </button>
    </div>
  </div>
</div>
`;

exports[`<Modal /> Snapshots should match snapshot with buttonsAligment prop equals center 1`] = `
.c9 {
  background-color: #c3c7cb;
  padding: 7px 20px 5px;
  border: none;
  font-size: 12px;
  box-shadow: inset 1px 1px 0px 1px #ffffff,inset 0 0 0 1px #868a8e,1px 1px 0 0px #000;
}

.c9:disabled {
  color: #868a8e;
}

.c9:focus {
  outline: none;
  box-shadow: inset 1px 1px 0px 1px #ffffff,inset -0.5px -0.5px 0px 1px #868a8e,1px 1px 0 1px #000;
  outline: 1px dotted #000;
  outline-offset: -5px;
}

.c9:active {
  padding: 8px 20px 5px;
  outline: 1px dotted #000;
  outline-offset: -5px;
  box-shadow: inset 0 0 0 1px #868a8e,0 0 0 1px #000;
}

.c2 {
  display: block;
  width: 15px;
  height: 13px;
  background-image: url('test-file-stub');
  background-repeat: no-repeat;
  background-position: center;
  background-size: contain;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: fixed;
  padding: 2px 2px 8px;
  width: 300px;
  height: 200px;
  top: 50px;
  background-color: #c3c7cb;
  box-shadow: inset 1px 1px 0px 1px #ffffff,inset 0 0 0 1px #868a8e,1px 1px 0 1px #000;
}

.c1 {
  height: 18px;
  margin-bottom: 2px;
  background-color: #00007f;
  color: white;
  padding: 2px 2px 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c4 {
  list-style: none;
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c6 {
  background-color: #c3c7cb;
  padding: 7px 20px 5px;
  border: none;
  font-size: 12px;
  box-shadow: inset 1px 1px 0px 1px #ffffff,inset 0 0 0 1px #868a8e,1px 1px 0 0px #000;
  margin-right: 2px;
  padding: 0;
  width: 17px;
  height: 14px;
  min-width: 0;
  font-size: 10px;
}

.c6:disabled {
  color: #868a8e;
}

.c6:focus {
  outline: none;
  box-shadow: inset 1px 1px 0px 1px #ffffff,inset -0.5px -0.5px 0px 1px #868a8e,1px 1px 0 1px #000;
  outline: 1px dotted #000;
  outline-offset: -5px;
}

.c6:active {
  padding: 8px 20px 5px;
  outline: 1px dotted #000;
  outline-offset: -5px;
  box-shadow: inset 0 0 0 1px #868a8e,0 0 0 1px #000;
}

.c6:last-child {
  margin-right: 0;
}

.c6:active {
  padding: 1px 0 0 1px;
  outline: none;
}

.c6:focus {
  box-shadow: inset 1px 1px 0px 1px #ffffff,inset -1px -1px 0px 1px #868a8e;
}

.c7 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 6px;
}

.c10 .c5 {
  margin-right: 6px;
  min-width: 70px;
}

.c10 .c5:last-child {
  margin-right: 0;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 6px 6px 6px;
}

.c8 .c5 {
  margin-right: 6px;
  min-width: 70px;
}

.c8 .c5:last-child {
  margin-right: 0;
}

<div>
  <div
    class="react-draggable c0"
    height="200"
    style="transform: translate(0px,0px);"
    width="300"
  >
    <div
      class="draggable c1"
    >
      <i
        class="c2"
        height="13"
        name="bat"
        style="margin-right: 4px;"
        width="15"
      />
      <div
        class="c3"
      >
        file.bat
      </div>
      <ul
        class="c4"
      >
        <button
          class="c5 c6"
        >
          ?
        </button>
        <button
          class="c5 c6"
        >
          x
        </button>
      </ul>
    </div>
    <div
      class="c7"
    >
      Hello
    </div>
    <div
      class="c8"
    >
      <button
        class="c5 c9"
        value="Ok"
      >
        Ok
      </button>
      <button
        class="c5 c9"
        value="Cancel"
      >
        Cancel
      </button>
    </div>
  </div>
</div>
`;

exports[`<Modal /> Snapshots should match snapshot with buttonsAligment prop equals flex-start 1`] = `
.c9 {
  background-color: #c3c7cb;
  padding: 7px 20px 5px;
  border: none;
  font-size: 12px;
  box-shadow: inset 1px 1px 0px 1px #ffffff,inset 0 0 0 1px #868a8e,1px 1px 0 0px #000;
}

.c9:disabled {
  color: #868a8e;
}

.c9:focus {
  outline: none;
  box-shadow: inset 1px 1px 0px 1px #ffffff,inset -0.5px -0.5px 0px 1px #868a8e,1px 1px 0 1px #000;
  outline: 1px dotted #000;
  outline-offset: -5px;
}

.c9:active {
  padding: 8px 20px 5px;
  outline: 1px dotted #000;
  outline-offset: -5px;
  box-shadow: inset 0 0 0 1px #868a8e,0 0 0 1px #000;
}

.c2 {
  display: block;
  width: 15px;
  height: 13px;
  background-image: url('test-file-stub');
  background-repeat: no-repeat;
  background-position: center;
  background-size: contain;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: fixed;
  padding: 2px 2px 8px;
  width: 300px;
  height: 200px;
  top: 50px;
  background-color: #c3c7cb;
  box-shadow: inset 1px 1px 0px 1px #ffffff,inset 0 0 0 1px #868a8e,1px 1px 0 1px #000;
}

.c1 {
  height: 18px;
  margin-bottom: 2px;
  background-color: #00007f;
  color: white;
  padding: 2px 2px 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c4 {
  list-style: none;
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c6 {
  background-color: #c3c7cb;
  padding: 7px 20px 5px;
  border: none;
  font-size: 12px;
  box-shadow: inset 1px 1px 0px 1px #ffffff,inset 0 0 0 1px #868a8e,1px 1px 0 0px #000;
  margin-right: 2px;
  padding: 0;
  width: 17px;
  height: 14px;
  min-width: 0;
  font-size: 10px;
}

.c6:disabled {
  color: #868a8e;
}

.c6:focus {
  outline: none;
  box-shadow: inset 1px 1px 0px 1px #ffffff,inset -0.5px -0.5px 0px 1px #868a8e,1px 1px 0 1px #000;
  outline: 1px dotted #000;
  outline-offset: -5px;
}

.c6:active {
  padding: 8px 20px 5px;
  outline: 1px dotted #000;
  outline-offset: -5px;
  box-shadow: inset 0 0 0 1px #868a8e,0 0 0 1px #000;
}

.c6:last-child {
  margin-right: 0;
}

.c6:active {
  padding: 1px 0 0 1px;
  outline: none;
}

.c6:focus {
  box-shadow: inset 1px 1px 0px 1px #ffffff,inset -1px -1px 0px 1px #868a8e;
}

.c7 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 6px;
}

.c10 .c5 {
  margin-right: 6px;
  min-width: 70px;
}

.c10 .c5:last-child {
  margin-right: 0;
}

.c11 .c5 {
  margin-right: 6px;
  min-width: 70px;
}

.c11 .c5:last-child {
  margin-right: 0;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  padding: 0 6px 6px 6px;
}

.c8 .c5 {
  margin-right: 6px;
  min-width: 70px;
}

.c8 .c5:last-child {
  margin-right: 0;
}

<div>
  <div
    class="react-draggable c0"
    height="200"
    style="transform: translate(0px,0px);"
    width="300"
  >
    <div
      class="draggable c1"
    >
      <i
        class="c2"
        height="13"
        name="bat"
        style="margin-right: 4px;"
        width="15"
      />
      <div
        class="c3"
      >
        file.bat
      </div>
      <ul
        class="c4"
      >
        <button
          class="c5 c6"
        >
          ?
        </button>
        <button
          class="c5 c6"
        >
          x
        </button>
      </ul>
    </div>
    <div
      class="c7"
    >
      Hello
    </div>
    <div
      class="c8"
    >
      <button
        class="c5 c9"
        value="Ok"
      >
        Ok
      </button>
      <button
        class="c5 c9"
        value="Cancel"
      >
        Cancel
      </button>
    </div>
  </div>
</div>
`;
